### ISSUE TITLE
Search component accessibility text missing

### ISSUE THREAD

**Issue by victoriadeanton** [Sage/carbon#6007]
### Current behaviour

Running Axe devTools for accessibility, in the Search component the button is missing a label.

![Screenshot 2023-05-09 at 08 33 55](https://user-images.githubusercontent.com/85164705/237013604-e548b8fe-6f66-48ea-82b4-a46f1b979502.png)

### Expected behaviour

The search button inside the Search component should have a label for the search button itself 

### CodeSandbox or Storybook URL

https://codesandbox.io/s/magical-currying-u7h37p

### JIRA Ticket (Sage Only)

SBS-57838

### Carbon Version

118.2.1

### What browsers are you seeing the problem on?

Chrome

### What Operating System are you seeing the problem on?

MacOS

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.
